java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example09_FunctionTypes.java

```diff
@@ -164,42 +174,103 @@ public String toString() {
                     .build(),
                 String.class)
             .block();
+        System.out.println(result.getResult());
 
-        kernel
+        result = kernel
             .invokeAsync(plugin.get("InputStringTaskWithStringResult"),
                 KernelArguments
                     .builder()
                     .withVariable("echoInput", "return this")
                     .build(),
                 String.class)
             .block();
+        System.out.println(result.getResult());
 
-        kernel
+        result = kernel
             .invokeAsync(plugin.get("InputStringTaskWithVoidResult"),
                 KernelArguments
                     .builder()
                     .withVariable("x", "x input")
                     .build(),
                 Void.class)
             .block();
+        System.out.println(result.getResult());
 
-        kernel
+        result = kernel
             .invokeAsync(plugin.get("noInputComplexReturnTypeAsync"),
                 null,
                 Temporal.class)
             .block();
+        System.out.println(result.getResult());
 
-        kernel
+        result = kernel
             .invokeAsync(plugin.get("noInputComplexReturnType"),
                 null,
                 Temporal.class)
             .block();
+        System.out.println(result.getResult());
 
-        kernel
+        // Possibilities for return type combinations:
+        // | Method return type | Declared Function Return Type | Invocation Return type |
+        // |--------------------|-------------------------------|------------------------|
+        // | T                  | T                             | T                      |
+        // | T                  | T                             | V converted from T     |
+
+        // | T                  | U extends T                   | T                      | This effectively requires a downcast...should we support this?
+        // | T                  | U extends T                   | U                      |
+        // | T                  | U extends T                   | V converted from U     |
+
+        result = kernel
+            .invokeAsync(plugin.get("conversionScenarioA"),
+                null,
+                OffsetDateTime.class)
+            .block();
+        System.out.println(result.getResult());
+
+        result = kernel
+            .invokeAsync(plugin.get("conversionScenarioA"),
+                null,
+                Instant.class)
+            .block();
+        System.out.println(result.getResult());
+
+        // Potentially unsupported scenario as it requires a downcast
+        /*
+        x = kernel
```

Comment:
What about a downcast method in the type converter? `<U extends T> Function<T, U> downcast`. Default implementation would just throw a SKException variant of a ClassCastException. There are some things that are easy to downcast, Number to Integer or Collection to List for example, but it obviously isn't an inline cast.